onmt/modules/Sequencer.lua

```diff
@@ -26,6 +28,16 @@ end
 function Sequencer:_sharedClone()
   local clone = self.network:clone('weight', 'gradWeight', 'bias', 'gradBias')
 
+  local cloneRNN = onmt.utils.Tensor.findModule(clone, self.rnnClass)
+  if cloneRNN ~= nil then
+    cloneRNN:shareRecurrents(onmt.utils.Tensor.findModule(self.network, self.rnnClass))
+  end
+  function clone:setTimeStep(t)
```

Comment:
I think it would be better to just build an array of all RNNs (in `self.rnns`) and define this function statically in the `Sequencer` class.

Also `_sharedClone` is not called at test time so `Sequencer.setTimeStep` will be undefined.